docs/source/quickstart.rst

```diff
@@ -1,2 +1,157 @@
+
+- `update_function`: a function which is passed a batch and passes data through and updates your model
+
+We are using helper methods :meth:`create_supervised_trainer` and :meth:`create_supervised_evaluator`:
+
+.. code-block:: python
+
+    trainer = create_supervised_trainer(model, optimizer, loss)
+    evaluator = create_supervised_evaluator(model,
+                                            metrics={
+                                                'accuracy': CategoricalAccuracy(),
+                                                'nll': Loss(loss)
+                                            })
+
+However, we could also define trainer and evaluator using :class:`Engine`. If we look into the code of
+:meth:`create_supervised_trainer` and :meth:`create_supervised_evaluator`, we can observe a pattern:
+
+.. code-block:: python
+
+    def create_engine(*args, **kwargs):
+
+        def _update(engine, batch):
+            # Update function logic
+            pass
+
+        return Engine(_update)
+
+And update functions of the trainer and evaluator are simply:
+
+.. code-block:: python
+
+    def _update(engine, batch):
+        model.train()
+        optimizer.zero_grad()
+        x, y = _prepare_batch(batch, cuda)
+        y_pred = model(x)
+        loss = loss_fn(y_pred, y)
+        loss.backward()
+        optimizer.step()
+        return loss.data.cpu()[0]
+
+    def _inference(engine, batch):
+        model.eval()
+        x, y = _prepare_batch(batch, cuda, volatile=True)
+        y_pred = model(x)
+        return to_tensor(y_pred, cpu=not cuda), to_tensor(y, cpu=not cuda)
+
+Remark that the helper function :meth:`create_supervised_evaluator` to create an evaluator accepts an
+argument `metrics`:
+
+.. code-block:: python
+
+    metrics={
+        'accuracy': CategoricalAccuracy(),
+        'nll': Loss(loss)
+    }
+
+where we define two metrics: *categorical accuracy* and *loss* to compute on validation dataset. More information on
+metrics can be found at :doc:`metrics`.
+
+
+The most interesting part of the code snippet is adding event handlers. :class:`Engine` allows to add handlers on
+various events that fired during the run. When event is fired, attached handlers (functions) are executed. Thus, for
+logging purposes we added a function to be executed after every iteration:
+
+.. code-block:: python
+
+    @trainer.on(Events.ITERATION_COMPLETED)
+    def log_training_loss(engine):
+        print("Epoch[{}] Loss: {:.2f}".format(engine.state.epoch, len(train_loader), engine.state.output))
+
+or equivalently without the decorator
+
+.. code-block:: python
+
+    def log_training_loss(engine):
+        print("Epoch[{}] Loss: {:.2f}".format(engine.state.epoch, len(train_loader), engine.state.output))
+
+    trainer.add_event_handler(Events.ITERATION_COMPLETED, log_training_loss)
+
+When an epoch ends we want to run model validation, therefore we attach another handler to the trainer on epoch complete
+event:
+
```

Comment:
nit: could put a note about passing other args to handlers. Or leave this for another doc/ official docstrings for `add_event_handler`